api/src/test/java/org/openmrs/notification/db/hibernate/HibernateAlertDAOTest.java

```diff
@@ -9,34 +9,68 @@
  */
 package org.openmrs.notification.db.hibernate;
 
+import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
+import org.openmrs.User;
 import org.openmrs.api.context.Context;
-import org.openmrs.notification.AlertService;
+import org.openmrs.notification.Alert;
 import org.openmrs.test.BaseContextSensitiveTest;
+import org.springframework.beans.factory.annotation.Autowired;
 
 /**
  * This class tests the hibernate alert data access. TODO Consider changing this and all subsequent
  * tests to use dbunit
  */
 public class HibernateAlertDAOTest extends BaseContextSensitiveTest {
 	
+	private static final String DATA_XML = "org/openmrs/api/db/hibernate/include/HibernateAlertDAOTestDataSet.xml";
+	@Autowired
+	private HibernateAlertDAO hibernateAlertDAO;
 	@Before
-	public void runBeforeEachTest() {
-		authenticate();
+
+	public void setUp() {
+		executeDataSet(DATA_XML);
+		
 	}
-	
 	/**
 	 * Test that you can get alerts
 	 * 
 	 * @throws Exception
 	 */
+	
 	@Test
-	public void shouldGetAlerts() {
+	public void saveAlert_shouldSaveAlertToDb() {
+		Alert saveAlert = hibernateAlertDAO.getAlert(2);
+	    hibernateAlertDAO.saveAlert(saveAlert);
+	    Assert.assertNotNull(saveAlert.getText());
+	}
+	
+	@Test
+	public void getAlert_shouldGetAlertById() {
+		Alert savedAlert = hibernateAlertDAO.getAlert(2);
+		int id=savedAlert.getAlertId();
+		Assert.assertEquals(id,2);
 		
-		AlertService as = Context.getAlertService();
-		//System.out.println(as.getAllAlerts());
+	}
+	@Test
+	public void deleteAlert_shouldReturnNullAfterDeleting() {
```

Comment:
@ibacher   hi ian,i finally got the delete method working,thanks very much for the reviews.

Reply:
@ibacher could it be looking good for merging?